### ISSUE TITLE
Bug in Performance/Count

### ISSUE THREAD

**Issue by bquorning** [bbatsov/rubocop#1824]
On master (6f216e5601b4cac97c95670e4daf0d7518fe753e), this code

```rb
Data = Struct.new(:value)

array = [Data.new(2), Data.new(3), Data.new(2)]

puts array.count
puts array.select(&:value).count
puts array.select(&:value).uniq.count
```

gives me this warning:
```
test.rb:7:33: C: Use count instead of select...count.
puts array.select(&:value).uniq.count
                                ^^^^^
```
Interestingly, I don’t get a warning on the somewhat similar line 6.

cc @rrosenblum

**Comment by bbatsov** [comment]
Btw, `count` without arguments should normally be replaced with `size`.

**Comment by bquorning** [comment]
> `count` without arguments should normally be replaced with `size`

True. Do we have a cop for that?

**Comment by bbatsov** [comment]
I don't think we do.

On 23 April 2015 at 14:40, Benjamin Quorning <notifications@github.com>
wrote:

> count without arguments should normally be replaced with size
>
> True. Do we have a cop for that?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bbatsov/rubocop/issues/1824#issuecomment-95557575>.
>



-- 
Best Regards,
Bozhidar Batsov

http://www.batsov.com

**Comment by rrosenblum** [comment]
Yes, this is a bug. I am a bit surprised that this is being registered as an offense because the cop is supposed to be looking for `count` called on `select`. I am sure I will learn something interesting in fixing this bug.

The `Performance/Size` cop will look for usages of `count` without a block and converts it to used `size`. The cop isn't very effective at the moment because it only complains under very specific circumstances. I was worried that `count` by itself was too generic of a method name to throw an offense based on the method by itself without something else to key on as well. There are a few improvements that need to be made to `Performance/Size`.
